docker-compose.yml

```diff
@@ -581,6 +581,9 @@ services:
       - redis
       - nginx-proxy:error-handler.dev.gov.uk
     environment:
+      PROXY_PERCENTAGE_OF_ASSET_REQUESTS_TO_S3_VIA_NGINX: 100
+      PROXY_PERCENTAGE_OF_WHITEHALL_ASSET_REQUESTS_TO_S3_VIA_NGINX: 100
+      FAKE_S3_HOST: http://127.0.0.1
```

Comment:
Asset Manager _proxies_ (rather than redirects) the request to the fake S3 URL (i.e. http://127.0.0.1/*) which is then served by a 2nd request to the Rails app. Because the request is proxied, the user does not see the fake S3 URL - it's just an implementation detail.

You are correct that the user cannot access the 127.0.0.1 address.

The fake S3 functionality is made up of some code in the Rails app which [saves files to a separate directory on the local filesystem](https://github.com/alphagov/asset-manager/blob/7e8c188a989d711d0f089d4eacd69a27cd4012f7/lib/s3_storage/fake.rb#L8-L13) and adds [an extra Rails route](https://github.com/alphagov/asset-manager/blob/7e8c188a989d711d0f089d4eacd69a27cd4012f7/config/routes.rb#L14-L16) which serves files directly from this separate directory.

Reply:
Cool thanks for explaining. I'm still a little confused but will address that as a separate comment